c7n/resources/s3.py

```diff
@@ -3402,11 +3402,18 @@ class BucketEncryption(KMSKeyResolverMixin, Filter):
                 filters:
                   - type: bucket-encryption
                     state: False
+              - name: s3-bucket-test-bucket-key-enabled
+                resource: s3
+                region: us-east-1
+                filters:
+                  - type: bucket-encryption
+                    bucket_key_enabled: True
     """
     schema = type_schema('bucket-encryption',
                          state={'type': 'boolean'},
                          crypto={'type': 'string', 'enum': ['AES256', 'aws:kms']},
-                         key={'type': 'string'})
+                         key={'type': 'string'},
+                         bucket_key_enabled={'type': 'boolean'})
```

Comment:
this probably needs a python validate impl or additional filtering functionality, atm if bucket_key_enabled and key is set, then the later is ignored, ie previous default behavior becomes a noop when using new.

Reply:
@kapilt I added a validation function to raise an error if bucket_key_enabled and key are both set. If you think it would be better to add a new filter for the BucketKeyEnabled check instead, please let me know.